### ISSUE TITLE
How to cancel de previous execution?

### ISSUE THREAD

**Issue by LorhanSohaky** [alewin/useWorker#118]
I'm trying to use fuzzy search in a worker, but I'm having the following error:
> [useWorker] You can only run one instance of the worker at a time, if you want to run more than one in parallel, create another instance with the hook useWorker()

Code:
```js
import debounce from 'lodash.debounce'
import { useWorker, WORKER_STATUS } from "@koale/useworker";

const fuseSearch = ({ list, query, limit, options }) => {
  // my fuse search
}

const MyComponent = () => {
  const [search, controller] = useWorker(fuseSearch)
  
  const handleInputChange = debounce(async value => {
    controller.kill()
    
    setLoading(true)
    try {
      const data = await search({ list: fundos, query: value, limit: 10, options })
      setLoading(false)
      setResultados(data)
    } catch (err) {
      if (err?.message) {
        setLoading(false)
        console.error(err)
      }
    }
  }, 300)
}
```

**Comment by ashish-r** [comment]
There is n issue with the kill functionality of the library, I have created a fix for the same. Waiting for the maintainers to review and merge the same.
https://github.com/alewin/useWorker/pull/122

**Comment by alewin** [comment]
released [`useWorker@4.1.0`](https://www.npmjs.com/package/@koale/useworker) with the fixes ( https://github.com/alewin/useWorker/pull/122 )


Apologies for the long wait, but due to personal and work-related reasons, I wasn’t able to maintain the library.
I’m well aware that forks and alternatives have emerged in the meantime, and I’m glad for it because the library was intentionally open-source with an MIT license to prevent issues like this.

Together with the new maintainer, @naorpeled, we are working to stabilize useWorker and add new features in the coming months. :)
